test/built-ins/RegExp/unicode_restricted_identity_escape.js

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2015 André Bargull. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: Extension B.1.4 is not applied for Unicode RegExp
+info: >
+    The compatibility extensions defined in B.1.4 Regular Expressions Patterns
+    are not applied for Unicode RegExps.
+    Tested extension: "IdentityEscape[U] :: [~U] SourceCharacter but not c"
+es6id: 21.1.2
+---*/
+
+function isSyntaxCharacter(c) {
+  switch (c) {
+    case "^":
+    case "$":
+    case "\\":
+    case ".":
+    case "*":
+    case "+":
+    case "?":
+    case "(":
+    case ")":
+    case "[":
+    case "]":
+    case "{":
+    case "}":
+    case "|":
+      return true;
+    default:
+      return false;
+  }
+}
+
+function isAlphaDigit(c) {
+  return ("0" <= c && c <= "9") || ("A" <= c && c <= "Z") || ("a" <= c && c <= "z");
+}
+
+
+// IdentityEscape in AtomEscape.
+//
+// AtomEscape[U] :: CharacterEscape[?U]
+// CharacterEscape[U] :: IdentityEscape[?U]
+for (var cu = 0x00; cu <= 0x7f; ++cu) {
+  var s = String.fromCharCode(cu);
+  if (!isAlphaDigit(s) && !isSyntaxCharacter(s) && s !== "/") {
+    assert.throws(SyntaxError, function() {
+      RegExp("\\" + s, "u");
+    }, "Invalid IdentityEscape '\\" + s + "'");
+  }
```

Comment:
The current if-condition is quite permissive, for example it skips over `A` because a future ES edition might add the [`\A` assertion](http://perldoc.perl.org/perlre.html#Assertions).

I could add two more test files:
1. Valid IdentityEscape in Unicode RegExp (`\<SyntaxCharacter>` and `\/`)
2. Invalid IdentityEscape (`\<Alpha>` except for known escapes like `\B` etc., that means a new test for the forbidden RegExp extension rule described in 16.1)